src/components/add-wine/useForm.js

```diff
@@ -0,0 +1,61 @@
+import {useState, useEffect, useCallback} from 'react';
+
+function useForm(stateSchema, validationSchema = {}, callback){
+  const [state, setState] = useState(stateSchema);
+  const [disable, setDisable] = useState(true);
+  const [isInvalid, setIsInvalid] = useState(true);
+
+  useEffect(() => {
+    setDisable(true)
+  }, []);
+
+  useEffect(()=>{
+    setDisable(validateState());
+  }, [state, isInvalid]);
+
+  const validateState = useCallback(() => {
+    const hasErrorInState = Object.keys(validationSchema).some(key => {
+      const isInputFieldRequired = validationSchema[key].required;
+      const stateValue = state[key].value;
+      const stateError = state[key].error;
+       return (isInputFieldRequired && !stateValue) || stateError;
+    })
+    return hasErrorInState;
+  }, [state, validationSchema]);
+
+  const handleOnChange = useCallback(event=>{
```

Comment:
I am not necessarily a fan of error messages popping up as the client is typing. Can the error messages be connected to the submit-button instead?

Reply:
Which means that the button doesn't need to be disabled, but that the validation is triggered on submit.

Reply:
done